### PR TITLE
Google Tasks node

### PR DESCRIPTION
**Pull request by Shraddha2104**
Hi,
I created a test node for Google Tasks API.
https://developers.google.com/tasks/v1/reference/tasks
Ran `npm run test` successfully

### CODE REVIEW

File: packages/nodes-base/nodes/Google/Calendar/GoogleCalendar.node.ts [hunk 1]
```diff
@@ -38,15 +28,15 @@ export class GoogleCalendar implements INodeType {
...
 		defaults: {
 			name: 'Google Calendar',
-			color: '#3E87E4',
+			color: '#3E87E4'
 		},
 		inputs: ['main'],
 		outputs: ['main'],
 		credentials: [
 			{
 				name: 'googleCalendarOAuth2Api',
```
Comment (ianjennings) [inline packages/nodes-base/nodes/Google/Calendar/GoogleCalendar.node.ts:37]: Is this an editor config thing? Does n8n have a suggested config? It would be easier to review without the formatting changes.

File: packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+						body.title = additionalFields.title as string;
+
+					if (additionalFields.dueDate)
+						body.dueDate = additionalFields.dueDate as string;
+
+					if (additionalFields.completed)
+						body.completed = additionalFields.completed as string;
+
+					if (additionalFields.deleted)
+						body.deleted = additionalFields.deleted as boolean;
```
Comment (ianjennings) [inline packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts:128]: Use brackets `{}` or shorthand `additionalFields.parent ? qs.parent = additionalFields.parent as string;`

See here:
https://stackoverflow.com/a/4797293/1757060

File: packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+
+					responseData = await googleApiRequest.call(
+						this,
+						'POST',
+						`/tasks/v1/lists/${taskId}/tasks`,
+						body,
+						qs
+					);
+				}
+				if (operation == 'delete') {
```
Comment (ianjennings) [inline packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts:147]: There are `==` here and `===` below.

File: packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+					body = {};
+					//https://developers.google.com/tasks/v1/reference/tasks/patch
+					const taskListId = this.getNodeParameter('task', i) as string;
+					const taskId = this.getNodeParameter('taskId', i) as string;
+					const updateFields = this.getNodeParameter(
+						'updateFields',
+						i
+					) as IDataObject;
+
+					if (updateFields.parent) qs.parent = updateFields.parent as string;
```
Comment (ianjennings) [inline packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts:239]: Same as above `{}`

File: packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+
+					if (updateFields.previous)
+						qs.previous = updateFields.previous as string;
+
+					if (updateFields.links) {
+						body.links = (updateFields.links as string[]).map(link => {
+							return { link: link };
```
Comment (ianjennings) [inline packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts:246]: You can do `return {link}`

File: packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts (hunk 2 above)
Comment (Shraddha2104) [reply packages/nodes-base/nodes/Google/Task/GoogleTasks.node.ts:128]: okay, I ll do that!!

File: packages/nodes-base/nodes/Google/Calendar/EventDescription.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { INodeProperties } from "n8n-workflow";
```
Comment (erin2722) [inline packages/nodes-base/nodes/Google/Calendar/EventDescription.ts:1]: Maybe we want to put these file changes into a different PR?

File: packages/nodes-base/nodes/Google/Task/GenericFunctions.ts [hunk 7]
```diff
@@ -0,0 +1,83 @@
...
+
+export async function googleApiRequest(
+	this: IExecuteFunctions | IExecuteSingleFunctions | ILoadOptionsFunctions,
+	method: string,
+	resource: string,
+	body: any = {},
+	qs: IDataObject = {},
+	uri?: string,
+	headers: IDataObject = {}
+): Promise<any> {
```
Comment (erin2722) [inline packages/nodes-base/nodes/Google/Task/GenericFunctions.ts:19]: As the code is written now, you might want to add in ` // tslint:disable-line:no-any` to get rid of the tslint errors associated with `<any>`

File: packages/nodes-base/nodes/Google/Task/GenericFunctions.ts [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+}
+
+export async function googleApiRequestAllItems(
+	this: IExecuteFunctions | ILoadOptionsFunctions,
+	propertyName: string,
+	method: string,
+	endpoint: string,
+	body: any = {},
+	query: IDataObject = {}
+): Promise<any> {
```
Comment (erin2722) [inline packages/nodes-base/nodes/Google/Task/GenericFunctions.ts:61]: Same here,  `// tslint:disable-line:no-any`

File: packages/nodes-base/nodes/MessageBird/MessageBird.node.ts [hunk 9]
```diff
@@ -1,17 +1,13 @@
-import {
```
Comment (erin2722) [inline packages/nodes-base/nodes/MessageBird/MessageBird.node.ts:1]: also might want this in a different branch

**Comment by erin2722** [comment]
Also, just checked out the task node. It looks good! Should the resource option be there if the user can only choose to operate on a task?

**Comment by Shraddha2104** [comment]
Merged against the n8n repo here: [n8n.io#680](https://github.com/n8n-io/n8n/pull/680)
